ydb/core/kqp/ut/service/kqp_qs_queries_ut.cpp

```diff
@@ -824,7 +824,255 @@ Y_UNIT_TEST_SUITE(KqpQueryService) {
+        TString Path;
+        THashMap<TString, TVector<TString>> Permissions;
+    };
+
+    Y_UNIT_TEST(DdlPermission) {
+        NKikimrConfig::TAppConfig appConfig;
+        appConfig.MutableTableServiceConfig()->SetEnablePreparedDdl(true);
+        auto setting = NKikimrKqp::TKqpSetting();
+        auto serverSettings = TKikimrSettings()
+            .SetAppConfig(appConfig)
+            .SetKqpSettings({setting});
+
+        TKikimrRunner kikimr(serverSettings);
+        auto db = kikimr.GetQueryClient();
+        auto session = kikimr.GetTableClient().CreateSession().GetValueSync().GetSession();
+
+        const auto checkPermissions = [](NYdb::NTable::TSession& session, TVector<ExpectedPermissions>&& expectedPermissionsValues) {
+            for (auto& value : expectedPermissionsValues) {
+                NYdb::NTable::TDescribeTableResult describe = session.DescribeTable(value.Path).GetValueSync();
+                UNIT_ASSERT_VALUES_EQUAL(describe.GetStatus(), EStatus::SUCCESS);
+                auto tableDesc = describe.GetTableDescription();
+                const auto& permissions = tableDesc.GetPermissions();
+
+                THashMap<TString, TVector<TString>> describePermissions;
+                for (const auto& permission : permissions) {
+                    auto& permissionNames = describePermissions[permission.Subject];
+                    permissionNames.insert(permissionNames.end(), permission.PermissionNames.begin(), permission.PermissionNames.end());
+                }
+
+                auto& expectedPermissions = value.Permissions;
+                UNIT_ASSERT_VALUES_EQUAL_C(expectedPermissions.size(), describePermissions.size(), "Number of user names does not equal on path: " + value.Path);
+                for (auto& item : expectedPermissions) {
+                    auto& expectedPermissionNames = item.second;
+                    auto& describedPermissionNames = describePermissions[item.first];
+                    UNIT_ASSERT_VALUES_EQUAL_C(expectedPermissionNames.size(), describedPermissionNames.size(), "Number of permissions for " + item.first + " does not equal on path: " + value.Path);
+                    sort(expectedPermissionNames.begin(), expectedPermissionNames.end());
+                    sort(describedPermissionNames.begin(), describedPermissionNames.end());
+                    UNIT_ASSERT_VALUES_EQUAL_C(expectedPermissionNames, describedPermissionNames, "Permissions are not equal on path: " + value.Path);
+                }
+            }
+        };
+
+        auto result = db.ExecuteQuery(R"(
+            GRANT ROW SELECT ON `/Root` TO user1;
+        )", TTxControl::NoTx()).ExtractValueSync();
+        UNIT_ASSERT_VALUES_EQUAL_C(result.GetStatus(), EStatus::GENERIC_ERROR, result.GetIssues().ToString());
+        UNIT_ASSERT_STRING_CONTAINS(result.GetIssues().ToString(), "Unexpected token 'ROW'");
+        checkPermissions(session, {{.Path = "/Root", .Permissions = {}}});
```

Comment:
Why empty?

Reply:
Because it is wrong requests and after them I expect that there aren't any permissions